Kitodo/src/main/java/de/sub/goobi/forms/ProzessverwaltungForm.java

```diff
@@ -206,14 +208,19 @@ public String saveAndRedirect() {
      */
     public String delete() {
         deleteMetadataDirectory();
-        this.process.getProject().getProcesses().remove(this.process);
         try {
-            Hibernate.initialize(process.getBatches());
-            for (Batch batch : process.getBatches()) {
+            this.process.getProject().getProcesses().remove(this.process);
+            this.process.setProject(null);
+            this.process.getTemplate().getProcesses().remove(this.process);
+            this.process.setTemplate(null);
+            List<Batch> batches = new CopyOnWriteArrayList<>(process.getBatches());
+            for (Batch batch : batches) {
                 batch.getProcesses().remove(this.process);
+                this.process.getBatches().remove(batch);
+                serviceManager.getBatchService().save(batch);
             }
             serviceManager.getProcessService().remove(this.process);
-        } catch (DataException e) {
+        } catch (DataException | RuntimeException e) {
```

Comment:
Catching a `RuntimeException` is really necessary here? Who is throwing this exception?

Reply:
Yes, it is necessary as if there happens some other problem e.g. NPE, application fails silently.